finite_difference/src/parallel/parallel_utils_mod.f90

```diff
@@ -120,6 +120,13 @@ end function get_rank
 
   !================================================
 
+  function on_master()
```

Comment:
I think you could define this routine just once in `parallel_mod` since, when running without MPI, get_rank() should always return 1? Please could you also add a brief docstring to the function.

Reply:
This is simpler of what I did before. Done.